fs.go

```diff
@@ -0,0 +1,73 @@
+package simpleblob
+
+import (
+	"bytes"
+	"context"
+	"io/fs"
+	"time"
+)
+
+// fsInterfaceWrapper wraps an Interface and implements fs.FS.
+type fsInterfaceWrapper struct{ Interface }
+
+// fsBlobWrapper represents data upstream and implements both fs.File
+// and fs.FileInfo for convenience.
+type fsBlobWrapper struct {
+	b      *Blob
+	parent *fsInterfaceWrapper
+	r      *bytes.Reader
+}
+
+// AsFS casts the provided interface to a fs.FS interface if supported,
+// else it wraps it to replicate its functionalities.
+func AsFS(st Interface) fs.FS {
```

Comment:
Since all storage methods require a context and the FS interface itself does have have a way to provide one, I suggest passing it here and holding on to it in the wrapper. Using `context.Background` somewhere other than main is a bad idea.